components/supervisor/pkg/supervisor/services.go

```diff
@@ -623,8 +624,11 @@ func (rt *remoteTokenProvider) GetToken(ctx context.Context, req *api.GetTokenRe
 
 // InfoService implements the api.InfoService.
 type InfoService struct {
-	cfg          *Config
-	ContentState ContentState
+	cfg            *Config
+	gitpodService  *gitpod.APIoverJSONRPC
+	tokenService   *InMemoryTokenService
+	workspaceClass *api.WorkspaceInfoResponse_WorkspaceClass
```

Comment:
should not it be protected by mutex to prevent concurrent requests to server?

Reply:
Maybe we should rather resolve it in supervisor.go and pass in services.go. It will be cleaner and does not need all this synchronisation.

Reply:
We can also consider to pass this info from env var from server on startup instead to avoid call at all.